app/models/concerns/filter/control_sections.rb

```diff
@@ -124,37 +124,37 @@ def control_sections
             id: 'coc_codes',
             label: 'CoC Codes',
             short_label: 'CoC',
-            value: @filter.chosen_coc_codes,
+            value: -> {  @filter.chosen_coc_codes },
           )
         end
         section.add_control(
           id: 'funding_sources',
-          value: @filter.funder_names,
+          value: -> { @filter.funder_names },
         )
         section.add_control(
           id: 'funding_other_sources',
-          value: @filter.funder_others,
+          value: -> { @filter.funder_others },
         )
         section.add_control(
           id: 'data_sources',
-          value: @filter.data_source_names,
+          value: -> { @filter.data_source_names },
         )
         section.add_control(
           id: 'organizations',
-          value: @filter.organization_names,
+          value: -> { @filter.organization_names },
         )
         section.add_control(
           id: 'projects',
-          value: @filter.project_names,
+          value: -> { @filter.project_names },
```

Comment:
This was the significant change but generally using lambdas for lazy evals seems to help the "Edit" report modal performance